engine-src/Game/LambdaHack/Common/Point.hs

```diff
@@ -131,6 +131,15 @@ adjacent s t = chessDist s t == 1
 -- not @PointI@, to permit aiming out of the level, e.g., to get
 -- uniform distributions of directions for explosions close to the edge
 -- of the level.
+-- |
+-- >>> bla 0 (Point 0 0) (Point 0 0) 
+-- Nothing
+-- >>> take 3 $ fromJust $ bla 0 (Point 0 0) (Point 1 0) 
+-- [(1,0),(2,0),(3,0)]
+-- >>> take 3 $ fromJust $ bla 0 (Point 0 0) (Point 0 1) 
+-- [(0,1),(0,2),(0,3)]
+-- >>> take 3 $ fromJust $ bla 0 (Point 0 0) (Point 1 1)
+-- [(1,1),(2,2),(3,3)]
 bla :: Int -> Point -> Point -> Maybe [Point]
```

Comment:
Feel free to rename the function and rename `blaXY` to the same with, perhaps, `Begin` suffix.